tests/msc3083_test.go

```diff
@@ -181,6 +184,130 @@ func TestRestrictedRoomsRemoteJoin(t *testing.T) {
 	checkRestrictedRoom(t, alice, bob, space, room)
 }
 
+// Not all servers can issue join events for a restricted room. Ensure a join
+// completes successfully when a server is in a room, but unable to issue joins
+// for it.
+func TestRestrictedRoomsLimitedJoin(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintFederationTwoLocalOneRemote)
+	defer deployment.Destroy(t)
+
+	// Charlie sets up the space so it is on the other server.
+	charlie := deployment.Client(t, "hs2", "@charlie:hs2")
+	space := charlie.CreateRoom(t, map[string]interface{}{
+		"preset": "public_chat",
+		"name":   "Space",
+		"creation_content": map[string]interface{}{
+			"type": "m.space",
+		},
+	})
+	// The room is an unstable room version which supports the restricted join_rule.
+	room := charlie.CreateRoom(t, map[string]interface{}{
+		"preset":       "public_chat",
+		"name":         "Room",
+		"room_version": "org.matrix.msc3083.v2",
+		"initial_state": []map[string]interface{}{
+			{
+				"type":      "m.room.join_rules",
+				"state_key": "",
+				"content": map[string]interface{}{
+					"join_rule": "restricted",
+					"allow": []map[string]interface{}{
+						{
+							"type":    "m.room_membership",
+							"room_id": &space,
+							"via":     []string{"hs2"},
+						},
+					},
+				},
+			},
+		},
+	})
+	charlie.SendEventSynced(t, space, b.Event{
+		Type:     "m.space.child",
+		StateKey: &room,
+		Content: map[string]interface{}{
+			"via": []string{"hs2"},
+		},
+	})
+
+	// Invite alice manually and accept it.
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+	charlie.InviteRoom(t, room, alice.UserID)
+	alice.JoinRoom(t, room, []string{"hs2"})
+
+	// Alice should not be able to issue invites.
```

Comment:
I assume that's because the default PLs in this room prevents her from doing so but could you add that info (or the real reason if I'm mistaken) to the comment? Since we're not _explicitly_ preventing her from inviting in the previous steps.

Reply:
Yes, I'm just confirming that the user cannot issue invites as expected! I'll make it clearer.